File: .vscode/striverSDE/CPP/arrays/left_rotate.cpp
#include<iostream>
using namespace std;

void left_rotate(int* arr, int n, int d) {
    int p = 1;
    while(p <= d) {
        int temp = arr[0];
        for(int i = 1; i < n; i++) {
            arr[i-1] = arr[i];
        }
        arr[n-1] = temp;
        p++;
    }
}

int main() {
    int arr1[] = {1, 3, 5, 2, 4};
    int n = sizeof(arr1) / sizeof(arr1[0]);
    int d = 3;
    left_rotate(arr1, n, d);
    for(int j = 0; j < n; j++) {
        cout << arr1[j] << " ";
    }
    return 0;
}

File: .vscode/striverSDE/Python/left_rotate.py
from typing import List

def left_rotated(arr: List[int],n,d)->int:
    p=0
    while(p<+d):    
      temp=arr[0]
      for i in range(1,n):
        arr[i-1]=arr[i]
      arr[n-1]=temp
      p+=1   



if __name__=="__main__" :

    arr1=[1,2,3,4,5]
    n=len(arr1)
    d=3
    left_rotated(arr1,n,d)
    for i in arr1:
        print(i,end="")